From 7c8c45dad3481cfaae3d3af9c188218aa14a3a6a Mon Sep 17 00:00:00 2001
From: Pavel Moravec <pmoravec@redhat.com>
Date: Tue, 9 Apr 2019 09:50:34 +0200
Subject: [PATCH] [foreman,satellite] increase plugin default timeouts

Those two plugins call commands with bigger timeouts than the default
plugin timeout is. That can unexpectedly kill the plugin execution when
the commands execution took longer than the plugin timeout (but within
cmd timeout).

Resolves: #1642

Signed-off-by: Pavel Moravec <pmoravec@redhat.com>
---
 sos/plugins/foreman.py   | 1 +
 sos/plugins/satellite.py | 1 +
 2 files changed, 2 insertions(+)

diff --git a/sos/plugins/foreman.py b/sos/plugins/foreman.py
index a1e937f3..8bcd26bd 100644
--- a/sos/plugins/foreman.py
+++ b/sos/plugins/foreman.py
@@ -19,6 +19,7 @@ class Foreman(Plugin):
     """
 
     plugin_name = 'foreman'
+    plugin_timeout = 1800
     profiles = ('sysmgmt',)
     packages = ('foreman', 'foreman-proxy')
     option_list = [
diff --git a/sos/plugins/satellite.py b/sos/plugins/satellite.py
index c50c2ec7..83733076 100644
--- a/sos/plugins/satellite.py
+++ b/sos/plugins/satellite.py
@@ -14,6 +14,7 @@ class Satellite(Plugin, RedHatPlugin):
     """
 
     plugin_name = 'satellite'
+    plugin_timeout = 1200
     profiles = ('sysmgmt',)
     verify_packages = ('spacewalk.*',)
     satellite = False
-- 
2.17.2

